platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/SwirldStateManagerUtils.java

```diff
@@ -52,7 +52,7 @@ public static MerkleRoot fastCopy(
 
         // Create a fast copy
         final MerkleRoot copy = state.copy();
-        final var platformState = copy.getPlatformState();
+        final var platformState = copy.getWritablePlatformState();
```

Comment:
Another idea for a future PR:
Split fast copy to readable/writable versions.  I have the impression (which may be incorrect) that a lot of fast copies are for purely read-only purposes, so perhaps adding a read-only version of this method would enable "safer" usage in those cases.

Reply:
Not sure if I fully understand how this change is supposed to look like. Could you please elaborate?

Reply:
Basically, create a `fastImmutableCopy` and a `fastMutableCopy` (perhaps `fastCopy` calls `fastMutableCopy` method initially).
Change calls, where appropriate, to call the `fastImmutableCopy` whenever possible so the calling component obtains an immutable state rather than a mutable one.

Not sure if that actually works, but it _seems_ like a possibly useful approach to reduce the number of mutable copies in the system.

Reply:
Ahh, thanks.  I was not aware the original object becomes immutable.
In that case, you're correct and my suggestion is moot.